plugins/copr.py

```diff
@@ -465,35 +467,32 @@ def _download_repo(self, project_name, repo_filename):
         arch = self.chroot_parts[-1]
         api_path = "/coprs/{0}/repo/{1}/dnf.repo?arch={2}".format(project_name, short_chroot, arch)
 
-        try:
-            f = self.base.urlopen(self.copr_url + api_path, mode='w+')
-        except IOError as e:
-            if os.path.exists(repo_filename):
-                os.remove(repo_filename)
-            if '404' in str(e):
-                if PY3:
-                    import urllib.request
-                    try:
-                        res = urllib.request.urlopen(self.copr_url + "/coprs/" + project_name)
-                        status_code = res.getcode()
-                    except urllib.error.HTTPError as e:
-                        status_code = e.getcode()
-                else:
-                    import urllib
-                    res = urllib.urlopen(self.copr_url + "/coprs/" + project_name)
+        response = requests.get(self.copr_url + api_path)
+        if response.status_code != 200:
+            if PY3:
+                import urllib.request
+                try:
+                    res = urllib.request.urlopen(self.copr_url + "/coprs/" + project_name)
                     status_code = res.getcode()
-                if str(status_code) != '404':
-                    raise dnf.exceptions.Error(_("This repository does not have"
-                                                 " any builds yet so you cannot enable it now."))
-                else:
-                    raise dnf.exceptions.Error(_("Such repository does not exist."))
-            raise
-
-        for line in f:
-            if re.match(r"\[copr:", line):
-                repo_filename = os.path.join(self.base.conf.get_reposdir,
-                                             "_" + line[1:-2] + ".repo")
-            break
+                except urllib.error.HTTPError as e:
+                    status_code = e.getcode()
+            else:
+                import urllib
+                res = urllib.urlopen(self.copr_url + "/coprs/" + project_name)
+                status_code = res.getcode()
+            if str(status_code) != '404':
+                msg = base64.b64decode(response.headers["Error-Message"]).decode('utf-8')
+                error_msg = ''.join(["It wasn't possible to enable this project.\n", msg])
+                raise dnf.exceptions.Error(_(error_msg))
+            else:
+                raise dnf.exceptions.Error(_("Such repository does not exist."))
+
+        if os.path.exists(repo_filename):
+            os.remove(repo_filename)
+        if re.match(r"\[copr:", response.text):
+            start = response.text.find("[")
+            end = response.text.find("]")
+            repo_filename = os.path.join(self.base.conf.get_reposdir, "_" + response.text[start+1:end] + ".repo")
```

Comment:
This is an unnecessarily invasive change, perhaps we need to split this to two or more commits to better explain why it is needed?

Reply:
It doesn't make sense to me to split it into multiple commits. The use of the re library is now applied to the entire text from the response.text, originally the f file obtained using self.base.urlopen was read line by line, now requests.get is used because I could not read the headers in the original version. Changes in the use of urlopen and requests are related.

Reply:
Well, thanks for the explanations but this is .... complicated.  This probably needs in-depth comment in commit message :-)  And at least I don't think `requests` is an acceptable dependency for the dnf-plugins-core package (you would have to put it into Requires: in the spec file).

So if headers are not readable using the dnf-internal downloader, I'd propose using the `urllib.urlopen()` (we already used that before), and using the url.readline() method (to keep the old line-by-line logic and to not change the code we don't have to change).